packages/contractkit/test/utils.ts

```diff
@@ -0,0 +1,44 @@
+interface MockContract {
+  methods: {
+    [methodName: string]: MockMethod
+  }
+  options: {
+    address: string
+  }
+}
+
+type MockMethod = (
+  ...params: any
+) => { call: () => any; estimateGas: () => number; send: SendMethod }
+
+type SendMethod = (...params: any) => { on: (...params: any) => any }
+
+const mockContractAddress = '0x000000000000000000000000000000000000CE10'
+
+export function createMockContract(methods: { [methodName: string]: any }) {
```

Comment:
This code looks familiar :)
How about we move it to a shared location to avoid the duplication? I don't love the idea of putting test related code in the `util` package but that's probably the best place for it atm

Reply:
Do you think we'll need this in the mobile code once it only interacts with contracts through contractkit? My understanding, based on a conversation with @ashishb a while ago, is that we'd be able to delete this from `mobile` once the migration of everything is complete.

However, if longterm, this code is needed in both places, I agree with you.

Reply:
I wouldn't recommend exporting this in utils. Utils is a dependency (not a dev one); and that would mean any project depending on utils will depend on jest and all it's dep graph.

Either we keep it here, or we create a test-utils package.

Reply:
@mcortesi great point, you're right, the utils package shouldn't export this.

@yerdua I'm not sure how the wallet code would leverage this from contract kit if we were to put it there. The kit shouldn't export it directly for the same reason mariano just stated above. We could try importing the source code directly from the mobile tests, but I'm not sure that'll work. Feel free to try.

I don't love the idea of making a whole other package just for this but it may be the least worst solution. @mcortesi any other ideas?

Reply:
@jmrossy What I meant by deleting it from mobile is that maybe mobile wouldn't need to mock contracts anymore, not that it would mock them using the code in this package. The ideal case, as I best understand it, is for users of the contractkit to not have to worry about the details of contracts themselves. Mobile/CLI/etc, might mock responses from `contractkit`, but not anything deeper.

Given that, I think it makes sense for this code to live here. If it is needed elsewhere, I'm in favor of a test-utils package and streamlining the process of managing packages.

Reply:
I think mobile may still need to mock because it's tests often need the contracts to return a specific fake value (e.g. testing when a user is half-verified)
If there would still be a way for mobile to do that without this util, then should be all good